taichi/backends/cuda/cuda_context.h

```diff
@@ -26,13 +27,21 @@ class CUDAContext {
   std::mutex lock;
   KernelProfilerBase *profiler;
   CUDADriver &driver;
+  CUDAProfiler &cuda_kernel_profiler;
```

Comment:
`_`

Reply:
Hmm would you mind explaining a bit why we have two profilers in CUDAcontext? It'd be nice if we can tell the difference from their name or adding a note :D

Reply:
`CUDAProfiler` has its own back-end methods that different from `KernelProfilerBase`.
This commit tries not to change `KernelProfilerBase`, so this new profiler class is added.
I'll think about how to better express and distinguish these two classes.

Reply:
Can we have  different implementation for `KernelProfilerBase` ? E.g. `CudaEventBasedProfiler`, `CuptiBasedProfiler`, etc?

Reply:
Removed from `CUDAContext`in commit 2eb03ed, CUPTI is now one of KernelProfilerCUDA's member variables, named `cupti_toolkit_`